alcs-api/src/application/application-subtask/application-subtask.service.spec.ts

```diff
@@ -0,0 +1,145 @@
+  let mockSubtaskRepo: DeepMocked<Repository<ApplicationSubtask>>;
+  let mockSubtaskTypeRepo: DeepMocked<Repository<ApplicationSubtaskType>>;
+  let applicationService: DeepMocked<ApplicationService>;
+
+  beforeEach(async () => {
+    applicationService = createMock<ApplicationService>();
+    mockSubtaskRepo = createMock<Repository<ApplicationSubtask>>();
+    mockSubtaskTypeRepo = createMock<Repository<ApplicationSubtaskType>>();
+
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [
+        ApplicationSubtaskService,
+        {
+          provide: ApplicationService,
+          useValue: applicationService,
+        },
+        {
+          provide: getRepositoryToken(ApplicationSubtask),
+          useValue: mockSubtaskRepo,
+        },
+        {
+          provide: getRepositoryToken(ApplicationSubtaskType),
+          useValue: mockSubtaskTypeRepo,
+        },
+      ],
+    }).compile();
+
+    applicationSubtaskService = module.get<ApplicationSubtaskService>(
+      ApplicationSubtaskService,
+    );
+  });
+
+  it('should be defined', () => {
+    expect(applicationSubtaskService).toBeDefined();
+  });
+
+  it('should check application exists and call through to repos for create', async () => {
+    const mockSubtask = {
+      uuid: 'fake-uuid',
+    } as ApplicationSubtask;
+    applicationService.get.mockResolvedValue({} as Application);
+    mockSubtaskTypeRepo.findOne.mockResolvedValue({} as ApplicationSubtaskType);
+    mockSubtaskRepo.save.mockResolvedValue({} as any);
+    mockSubtaskRepo.findOne.mockResolvedValue(mockSubtask);
+
+    const res = await applicationSubtaskService.create(
+      'mock-file',
+      'fake-type',
+    );
+
+    expect(applicationService.get).toHaveBeenCalled();
+    expect(mockSubtaskTypeRepo.findOne).toHaveBeenCalled();
+    expect(mockSubtaskRepo.save).toHaveBeenCalled();
+    expect(mockSubtaskRepo.findOne).toHaveBeenCalled();
+    expect(res).toEqual(mockSubtask);
+  });
+
+  it('should throw an exception if application doesnt exist for create', async () => {
+    applicationService.get.mockResolvedValue(undefined);
+
+    await expect(
+      applicationSubtaskService.create('mock-file', 'fake-type'),
+    ).rejects.toMatchObject(
+      new ServiceNotFoundException(`File number not found mock-file`),
+    );
+
+    expect(applicationService.get).toHaveBeenCalled();
+  });
+
+  it('should throw an exception if type does not exist', async () => {
+    applicationService.get.mockResolvedValue({} as Application);
+    mockSubtaskTypeRepo.findOne.mockResolvedValue(undefined);
+
+    await expect(
+      applicationSubtaskService.create('mock-file', 'fake-type'),
+    ).rejects.toMatchObject(
+      new ServiceNotFoundException(`Invalid subtask type fake-type`),
+    );
+
+    expect(applicationService.get).toHaveBeenCalled();
+    expect(mockSubtaskTypeRepo.findOne).toHaveBeenCalled();
+  });
+
+  it('should call save and set fields for update', async () => {
```

Comment:
Is there any case where the createdAt needs to change (or should potentially be startedAt)?